src/component/menu/ApiDetails/ApiDetails.tsx

```diff
@@ -0,0 +1,106 @@
+import useUiConfig from '../../../hooks/api/getters/useUiConfig/useUiConfig';
+import { ReactElement } from 'react';
+import ConditionallyRender from '../../common/ConditionallyRender';
+import { IVersionInfo } from '../../../interfaces/uiConfig';
+
+interface IPartialUiConfig {
+	name: string;
+	version: string;
+	slogan?: string;
+	environment?: string;
+	versionInfo?: IVersionInfo;
+}
+
+interface IProps {
+	uiConfig: IPartialUiConfig;
+}
+
+export function ApiDetailsContainer(): ReactElement {
+	const { uiConfig } = useUiConfig();
+
+	return <ApiDetails uiConfig={uiConfig} />;
+}
+
+export function ApiDetails(props: IProps): ReactElement {
+	return (
+		<section title="API details">
+			<CurrentVersion {...props} />
+			<UpdateNotification {...props} />
+			<br />
+			<small>{props.uiConfig.slogan}</small>
+			<br />
+			<InstanceId {...props} />
+		</section>
+	);
+}
+
+function InstanceId(props: IProps) {
+	const instanceId = props.uiConfig.versionInfo?.instanceId;
+
+	return (
+		<ConditionallyRender
+			condition={instanceId}
+			show={<small>{`${instanceId}`}</small>}
+		/>
+	);
+}
+
+function CurrentVersion(props: IProps) {
+	const currentVersion = formatCurrentVersion(props.uiConfig);
+	const environment = props.uiConfig.environment;
+
+	return (
+		<h4>
+			{currentVersion}{' '}
+			<ConditionallyRender
+				condition={environment}
+				show={<small>({environment})</small>}
+			/>
+		</h4>
+	);
+}
+
+function UpdateNotification(props: IProps) {
+	const updateNotification = formatUpdateNotification(props.uiConfig);
+
+	return (
+		<ConditionallyRender
+			condition={updateNotification}
+			show={
+				<small>
+					{updateNotification}
+					<br />
+				</small>
+			}
+		/>
+	);
+}
+
+function formatCurrentVersion(uiConfig: IPartialUiConfig): string {
```

Comment:
I usually place these in helper files inside of the component directory. It's not a must, but I feel good knowing that a component file just keeps the component and then the directory contains related functionality: 

```
-- ApiDetails
      -- ApiDetails.tsx
      -- helpers.ts
```
   

I'm open to listen to suggestions, but I feel it's sometimes easier to understand a file if I remove the clutter from the component and I find that if it's imported it's easier to locate rather than having them at the bottom of the file.